common/cache/lru.go

```diff
@@ -41,13 +45,15 @@ var (
 // lru is a concurrent fixed size cache that evicts elements in lru order
 type (
 	lru struct {
-		mut      sync.Mutex
-		byAccess *list.List
-		byKey    map[interface{}]*list.Element
-		maxSize  int
-		ttl      time.Duration
-		pin      bool
-		clock    clockwork.Clock
+		mut           sync.Mutex
+		byAccess      *list.List
+		byKey         map[interface{}]*list.Element
+		maxSize       int
+		maxTotalBytes int
+		totalBytes    int
```

Comment:
can we use one unit?
this unit can be user defined, either size or count

Reply:
This change is on top of existing count base cache. Is up to we prefer allows the cache has 2 limits at same time or single limit. @yycptt do you have a preference?